frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetailWrapper.tsx

```diff
@@ -28,10 +29,15 @@ export function ObjectDetailWrapper({
         className={""} // need an empty className to override the Modal default width
       >
         <ObjectDetailView
-          {...props}
+          {...rest}
           showHeader
           data={data}
-          question={question}
+          question={
+            question ??
+            (card && rest.metadata
```

Comment:
there is a case when ObjectDetail is displayed on a dashboard, in this case `question` from redux is not defined, we're still able to get it.

Another case (which is covered by optional params) is showing a public question, I need to double check if metadata is available in that case